pkg/auth/iam/iam.go

```diff
@@ -63,6 +63,9 @@ func NewFilter(client iam.Client) *Filter {
 
 // NewFilterWithOptions creates new Filter instance with Options
 func NewFilterWithOptions(client iam.Client, options *FilterInitializationOptions) *Filter {
+	if options == nil {
+		return NewFilter(client)
```

Comment:
Ha, sorry, My opinion is to throw the error while the options is nil

because why the developer want to set nil, but not use above method?

Reply:
hm, it will make different return options with what the `NewFilter` do.
is it oke?

Reply:
yes, this is a problem. both are ok from my side. 
@fadhillahentino